server/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java

```diff
@@ -1048,7 +1048,10 @@ public void testFilterCacheStats() throws Exception {
         assertEquals(DocWriteResponse.Result.DELETED, client().prepareDelete("index", "type", "1").get().getResult());
         assertEquals(DocWriteResponse.Result.DELETED, client().prepareDelete("index", "type", "2").get().getResult());
         if (IndexSettings.INDEX_SOFT_DELETES_SETTING.get(settings)) {
-            persistGlobalCheckpoint("index"); // Need to persist the global checkpoint for the soft-deletes retention MP.
+            // Need to persist the global checkpoint and flush a new safe commit
+            // as the soft-deletes retention merge policy depends on the local-checkpoint of the safe commit.
+            persistGlobalCheckpoint("index");
```

Comment:
not sure about this I think @bleskes  needs to look at this

Reply:
I don't follow the comment - @dnhatn can you please explain?

Reply:
I added a comment for this.